### PR TITLE
Add minimal tests to ensure dialogs don't throw when created

### LINKED ISSUE #640: Review tests of Dialogs
Issue #638 showed some problems with dialogs failing to display - throwing an exception when they were activated by the user. At a minimum, we need some tests that display each dialog without an exception. Additional testing within each dialog is highly desired as well but this may require some restructuring of the app, since dialogs don't participate in our MVP architecture when displayed. Maybe something like the Humble Dialog pattern will work for this.

### PR DESCRIPTION
**Pull request by CharliePoole**
fixes #640

### CODE REVIEW

File: src/TestCentric/tests/Dialogs/DialogCreationTests.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        // it references the main presenter. This needs to be changed.
+        [TestCase(typeof(AboutBox))]
+        [TestCase(typeof(TestPropertiesDialog))]
+        [TestCase(typeof(SettingsDialogBase))]
+        [TestCase(typeof(ExtensionDialog), null)]
+        [TestCase(typeof(ParameterDialog))]
+        [TestCase(typeof(TestParametersDialog))]
+        public void CreateDialog(Type dlgType, params object[] args)
+        {
+            Activator.CreateInstance(dlgType, args);
```
Comment (immeraufdemhund) [inline src/TestCentric/tests/Dialogs/DialogCreationTests.cs:25]: I read a long time ago that having test code show that no errors are thrown is not a test. I'm just wondering if a slightly more verbose test wouldn't be in order. Like that no error is thrown or that the object isn't null?

Comment (CharliePoole) [reply src/TestCentric/tests/Dialogs/DialogCreationTests.cs:25]: I agree with that and it's definitely the received wisdom. I've even preached it myself. That's why there's a comment about it's being minimal. :smile: 

I could make it more formally a test by surrounding the creation with `Assert.DoesNotThrow` but I tried it both ways and I think the output from this one is easier to understand.

I could test for non-null, but I believe that's impossible unless you instantiate a `Nullable<T>` with `Activator`.

So bottom line... the errors you found would have been caught by these tests, so I have to say they __are__ real tests!

Future direction: make a separate test for each dialog, do the create in a OneTimeSetUp method and verify other stuff about the created dialog.

Further future: split each into two class using the Humble Dialog pattern. That will allow separate testing of the controlling class.

Comment (CharliePoole) [reply src/TestCentric/tests/Dialogs/DialogCreationTests.cs:25]: @immeraufdemhund But feel free to improve this after I merge!!! :smile_cat:
